src/main/java/components/Task.java

```diff
@@ -0,0 +1,82 @@
+package components;
+
+/**
+ * Class for Task object encapsulate commonalities in subclasses.
+ */
+public class Task {
+    private int done;
```

Comment:
I think there may introduce the magic number issue here. How about changing the toggle to a boolean type instead?